### ISSUE TITLE
Vagrant fails to build a war file

### ISSUE THREAD

**Issue by bricas** [IQSS/dataverse#2744]
I've checked out the v4.2.1 tag and tried to do a ```vagrant up```, but it keeps failing on building the .war file. I've add -X to the maven command and I'm getting this output:

```
    standalone: Running: /var/folders/5s/1bbktts17ps69l171hcgp54wq45dfb/T/vagrant-shell20151117-14753-no2a4q.sh
==> standalone: no war file found... building
==> standalone: Apache Maven 3.3.3 (7994120775791599e205a5524ec3e0dfe41d4a06; 2015-04-22T04:57:37-07:00)
==> standalone: Maven home: /usr/share/apache-maven
==> standalone: Java version: 1.7.0_91, vendor: Oracle Corporation
==> standalone: Java home: /usr/lib/jvm/java-1.7.0-openjdk-1.7.0.91.x86_64/jre
==> standalone: Default locale: en_US, platform encoding: UTF-8
==> standalone: OS name: "linux", version: "2.6.32-431.el6.x86_64", arch: "amd64", family: "unix"
==> standalone: [DEBUG] Created new class realm maven.api
==> standalone: [DEBUG] Importing foreign packages into class realm maven.api
==> standalone: [DEBUG]   Imported: javax.enterprise.inject.* < plexus.core
==> standalone: [DEBUG]   Imported: javax.enterprise.util.* < plexus.core
==> standalone: [DEBUG]   Imported: javax.inject.* < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.* < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.artifact < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.classrealm < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.cli < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.configuration < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.exception < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.execution < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.execution.scope < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.lifecycle < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.model < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.monitor < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.plugin < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.profiles < plexus.core
==> standalone:
==> standalone: [DEBUG]   Imported: org.apache.maven.project < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.reporting < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.repository < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.rtinfo < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.settings < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.toolchain < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.usability < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.wagon.* < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.wagon.authentication < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.wagon.authorization < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.wagon.events < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.wagon.observers < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.wagon.proxy < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.wagon.repository < plexus.core
==> standalone: [DEBUG]   Imported: org.apache.maven.wagon.resource < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.classworlds < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.* < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.classworlds < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.component < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.configuration < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.container < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.context < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.lifecycle < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.logging < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.personality < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.util.xml.Xpp3Dom < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.util.xml.pull.XmlPullParser < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.util.xml.pull.XmlPullParserException < plexus.core
==> standalone: [DEBUG]   Imported: org.codehaus.plexus.util.xml.pull.XmlSerializer < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.* < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.artifact < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.collection < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.deployment < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.graph < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.impl < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.installation < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.internal.impl < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.metadata < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.repository < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.resolution < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.spi < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.transfer < plexus.core
==> standalone: [DEBUG]   Imported: org.eclipse.aether.version < plexus.core
==> standalone: [DEBUG]   Imported: org.slf4j.* < plexus.core
==> standalone:
==> standalone: [DEBUG]   Imported: org.slf4j.spi.* < plexus.core
==> standalone: [DEBUG] Populating class realm maven.api
==> standalone: [INFO] Error stacktraces are turned on.
==> standalone: [DEBUG] Reading global settings from /usr/share/apache-maven/conf/settings.xml
==> standalone: [DEBUG] Reading user settings from /home/vagrant/.m2/settings.xml
==> standalone: [DEBUG] Reading global toolchains from /usr/share/apache-maven/conf/toolchains.xml
==> standalone: [DEBUG] Reading user toolchains from /home/vagrant/.m2/toolchains.xml
==> standalone: [DEBUG] Using local repository at /home/vagrant/.m2/repository
==> standalone: [DEBUG] Using manager EnhancedLocalRepositoryManager with priority 10.0 for /home/vagrant/.m2/repository
==> standalone: [INFO] Scanning for projects...
==> standalone: [DEBUG] Extension realms for project edu.harvard.iq:dataverse:war:4.2.1: (none)
==> standalone: [DEBUG] Looking up lifecyle mappings for packaging war from ClassRealm[plexus.core, parent: null]
==> standalone: [DEBUG] === REACTOR BUILD PLAN ================================================
==> standalone: [DEBUG] Project: edu.harvard.iq:dataverse:war:4.2.1
==> standalone: [DEBUG] Tasks:   [package]
==> standalone: [DEBUG] Style:   Regular
==> standalone:
==> standalone: [DEBUG] =======================================================================
==> standalone: [INFO]
==> standalone: [INFO] ------------------------------------------------------------------------
==> standalone: [INFO] Building dataverse 4.2.1
==> standalone: [INFO] ------------------------------------------------------------------------
==> standalone: [DEBUG] Lifecycle default -> [validate, initialize, generate-sources, process-sources, generate-resources, process-resources, compile, process-classes, generate-test-sources, process-test-sources, generate-test-resources, process-test-resources, test-compile, process-test-classes, test, prepare-package, package, pre-integration-test, integration-test, post-integration-test, verify, install, deploy]
==> standalone: [DEBUG] Lifecycle clean -> [pre-clean, clean, post-clean]
==> standalone: [DEBUG] Lifecycle site -> [pre-site, site, post-site, site-deploy]
==> standalone: [DEBUG] Using transporter WagonTransporter with priority -1.0 for https://repo.maven.apache.org/maven2
==> standalone: [DEBUG] Using connector BasicRepositoryConnector with priority 0.0 for https://repo.maven.apache.org/maven2
==> standalone: Downloading: https://repo.maven.apache.org/maven2/org/jacoco/jacoco-maven-plugin/0.7.5.201505241946/jacoco-maven-plugin-0.7.5.201505241946.pom
==> standalone: [DEBUG] Writing tracking file /home/vagrant/.m2/repository/org/jacoco/jacoco-maven-plugin/0.7.5.201505241946/jacoco-maven-plugin-0.7.5.201505241946.pom.lastUpdated
==> standalone:
==> standalone: [INFO] ------------------------------------------------------------------------
==> standalone: [INFO] BUILD FAILURE
==> standalone: [INFO] ------------------------------------------------------------------------
==> standalone: [INFO] Total time: 2.847 s
==> standalone: [INFO] Finished at: 2015-11-17T12:21:09-08:00
==> standalone: [INFO] Final Memory: 7M/28M
==> standalone: [INFO] ------------------------------------------------------------------------
==> standalone: [ERROR] Plugin org.jacoco:jacoco-maven-plugin:0.7.5.201505241946 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.jacoco:jacoco-maven-plugin:jar:0.7.5.201505241946: Could not transfer artifact org.jacoco:jacoco-maven-plugin:pom:0.7.5.201505241946 from/to central (https://repo.maven.apache.org/maven2): java.security.ProviderException: java.security.KeyException -> [Help 1]
==> standalone: org.apache.maven.plugin.PluginResolutionException: Plugin org.jacoco:jacoco-maven-plugin:0.7.5.201505241946 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.jacoco:jacoco-maven-plugin:jar:0.7.5.201505241946
==> standalone: 	at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolve(DefaultPluginDependenciesResolver.java:117)
==> standalone: 	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getPluginDescriptor(DefaultMavenPluginManager.java:179)
==> standalone: 	at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getMojoDescriptor(DefaultMavenPluginManager.java:298)
==> standalone: 	at org.apache.maven.plugin.DefaultBuildPluginManager.getMojoDescriptor(DefaultBuildPluginManager.java:241)
==> standalone: 	at org.apache.maven.lifecycle.internal.DefaultLifecycleMappingDelegate.calculateLifecycleMappings(DefaultLifecycleMappingDelegate.java:110)
==> standalone: 	at org.apache.maven.lifecycle.internal.DefaultLifecycleExecutionPlanCalculator.calculateLifecycleMappings(DefaultLifecycleExecutionPlanCalculator.java:266)
==> standalone: 	at org.apache.maven.lifecycle.internal.DefaultLifecycleExecutionPlanCalculator.calculateMojoExecutions(DefaultLifecycleExecutionPlanCalculator.java:217)
==> standalone: 	at org.apache.maven.lifecycle.internal.DefaultLifecycleExecutionPlanCalculator.calculateExecutionPlan(DefaultLifecycleExecutionPlanCalculator.java:127)
==> standalone: 	at org.apache.maven.lifecycle.internal.DefaultLifecycleExecutionPlanCalculator.calculateExecutionPlan(DefaultLifecycleExecutionPlanCalculator.java:145)
==> standalone: 	at org.apache.maven.lifecycle.internal.builder.BuilderCommon.resolveBuildPlan(BuilderCommon.java:96)
==> standalone: 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:109)
==> standalone: 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
==> standalone: 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
==> standalone: 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
==> standalone: 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
==> standalone: 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
==> standalone: 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
==> standalone: 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
==> standalone: 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
==> standalone: 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
==> standalone: 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
==> standalone: 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
==> standalone: 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
==> standalone: 	at java.lang.reflect.Method.invoke(Method.java:606)
==> standalone: 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
==> standalone: 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
==> standalone: 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
==> standalone: 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
==> standalone: Caused by: org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for org.jacoco:jacoco-maven-plugin:jar:0.7.5.201505241946
==> standalone: 	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:302)
==> standalone: 	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.readArtifactDescriptor(DefaultArtifactDescriptorReader.java:218)
==> standalone: 	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.readArtifactDescriptor(DefaultRepositorySystem.java:287)
==> standalone: 	at org.apache.maven.plugin.internal.DefaultPluginDependenciesResolver.resolve(DefaultPluginDependenciesResolver.java:103)
==> standalone: 	... 27 more
==> standalone: Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not transfer artifact org.jacoco:jacoco-maven-plugin:pom:0.7.5.201505241946 from/to central (https://repo.maven.apache.org/maven2): java.security.ProviderException: java.security.KeyException
==> standalone: 	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:444)
==> standalone: 	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:246)
==> standalone: 	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:223)
==> standalone: 	at org.apache.maven.repository.internal.DefaultArtifactDescriptorReader.loadPom(DefaultArtifactDescriptorReader.java:287)
==> standalone: 	... 30 more
==> standalone: Caused by: org.eclipse.aether.transfer.ArtifactTransferException: Could not transfer artifact org.jacoco:jacoco-maven-plugin:pom:0.7.5.201505241946 from/to central (https://repo.maven.apache.org/maven2): java.security.ProviderException: java.security.KeyException
==> standalone: 	at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed(ArtifactTransportListener.java:43)
==> standalone: 	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:355)
==> standalone: 	at org.eclipse.aether.util.concurrency.RunnableErrorForwarder$1.run(RunnableErrorForwarder.java:67)
==> standalone: 	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$DirectExecutor.execute(BasicRepositoryConnector.java:581)
==> standalone: 	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get(BasicRepositoryConnector.java:249)
==> standalone: 	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:520)
==> standalone: 	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:421)
==> standalone: 	... 33 more
==> standalone: Caused by: org.apache.maven.wagon.TransferFailedException: java.security.ProviderException: java.security.KeyException
==> standalone: 	at org.apache.maven.wagon.providers.http.AbstractHttpClientWagon.fillInputData(AbstractHttpClientWagon.java:1066)
==> standalone: 	at org.apache.maven.wagon.providers.http.AbstractHttpClientWagon.fillInputData(AbstractHttpClientWagon.java:960)
==> standalone: 	at org.apache.maven.wagon.StreamWagon.getInputStream(StreamWagon.java:116)
==> standalone: 	at org.apache.maven.wagon.StreamWagon.getIfNewer(StreamWagon.java:88)
==> standalone: 	at org.apache.maven.wagon.StreamWagon.get(StreamWagon.java:61)
==> standalone: 	at org.eclipse.aether.transport.wagon.WagonTransporter$GetTaskRunner.run(WagonTransporter.java:560)
==> standalone: 	at org.eclipse.aether.transport.wagon.WagonTransporter.execute(WagonTransporter.java:427)
==> standalone: 	at org.eclipse.aether.transport.wagon.WagonTransporter.get(WagonTransporter.java:404)
==> standalone: 	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$GetTaskRunner.runTask(BasicRepositoryConnector.java:447)
==> standalone: 	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:350)
==> standalone: 	... 38 more
==> standalone: Caused by: javax.net.ssl.SSLException: java.security.ProviderException: java.security.KeyException
==> standalone: 	at sun.security.ssl.Alerts.getSSLException(Alerts.java:208)
==> standalone: 	at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1916)
==> standalone: 	at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1874)
==> standalone: 	at sun.security.ssl.SSLSocketImpl.handleException(SSLSocketImpl.java:1857)
==> standalone: 	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1378)
==> standalone: 	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1355)
==> standalone: 	at org.apache.maven.wagon.providers.http.httpclient.conn.ssl.SSLConnectionSocketFactory.createLayeredSocket(SSLConnectionSocketFactory.java:275)
==> standalone: 	at org.apache.maven.wagon.providers.http.httpclient.conn.ssl.SSLConnectionSocketFactory.connectSocket(SSLConnectionSocketFactory.java:254)
==> standalone: 	at org.apache.maven.wagon.providers.http.httpclient.impl.conn.HttpClientConnectionOperator.connect(HttpClientConnectionOperator.java:123)
==> standalone: 	at org.apache.maven.wagon.providers.http.httpclient.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:318)
==> standalone: 	at org.apache.maven.wagon.providers.http.httpclient.impl.execchain.MainClientExec.establishRoute(MainClientExec.java:363)
==> standalone: 	at org.apache.maven.wagon.providers.http.httpclient.impl.execchain.MainClientExec.execute(MainClientExec.java:219)
==> standalone: 	at org.apache.maven.wagon.providers.http.httpclient.impl.execchain.ProtocolExec.execute(ProtocolExec.java:195)
==> standalone: 	at org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RetryExec.execute(RetryExec.java:86)
==> standalone: 	at org.apache.maven.wagon.providers.http.httpclient.impl.execchain.RedirectExec.execute(RedirectExec.java:108)
==> standalone: 	at org.apache.maven.wagon.providers.http.httpclient.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
==> standalone: 	at org.apache.maven.wagon.providers.http.httpclient.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:82)
==> standalone: 	at org.apache.maven.wagon.providers.http.AbstractHttpClientWagon.execute(AbstractHttpClientWagon.java:832)
==> standalone: 	at org.apache.maven.wagon.providers.http.AbstractHttpClientWagon.fillInputData(AbstractHttpClientWagon.java:983)
==> standalone: 	... 47 more
==> standalone: Caused by: java.security.ProviderException: java.security.KeyException
==> standalone: 	at sun.security.ec.ECKeyPairGenerator.generateKeyPair(ECKeyPairGenerator.java:146)
==> standalone: 	at java.security.KeyPairGenerator$Delegate.generateKeyPair(KeyPairGenerator.java:704)
==> standalone: 	at sun.security.ssl.ECDHCrypt.<init>(ECDHCrypt.java:78)
==> standalone: 	at sun.security.ssl.ClientHandshaker.serverKeyExchange(ClientHandshaker.java:717)
==> standalone: 	at sun.security.ssl.ClientHandshaker.processMessage(ClientHandshaker.java:278)
==> standalone: 	at sun.security.ssl.Handshaker.processLoop(Handshaker.java:913)
==> standalone: 	at sun.security.ssl.Handshaker.process_record(Handshaker.java:849)
==> standalone: 	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1035)
==> standalone: 	at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1344)
==> standalone: 	at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:1371)
==> standalone: 	... 61 more
==> standalone: Caused by: java.security.KeyException
==> standalone: 	at sun.security.ec.ECKeyPairGenerator.generateECKeyPair(Native Method)
==> standalone: 	at sun.security.ec.ECKeyPairGenerator.generateKeyPair(ECKeyPairGenerator.java:126)
==> standalone: 	... 70 more
==> standalone: [ERROR]
==> standalone: [ERROR]
==> standalone: [ERROR] For more information about the errors and possible solutions, please read the following articles:
==> standalone: [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
```

The stange thing is that I can do

```
curl -L https://repo.maven.apache.org/maven2/org/jacoco/jacoco-maven-plugin/0.7.5.201505241946/jacoco-maven-plugin-0.7.5.201505241946.pom
```

and get this output:

```
<?xml version="1.0" encoding="UTF-8"?>
<!--
   Copyright (c) 2009, 2015 Mountainminds GmbH & Co. KG and Contributors
   All rights reserved. This program and the accompanying materials
   are made available under the terms of the Eclipse Public License v1.0
   which accompanies this distribution, and is available at
   http://www.eclipse.org/legal/epl-v10.html

   Contributors:
      Evgeny Mandrikov - initial API and implementation
-->
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <parent>
    <groupId>org.jacoco</groupId>
    <artifactId>org.jacoco.build</artifactId>
    <version>0.7.5.201505241946</version>
    <relativePath>../org.jacoco.build</relativePath>
  </parent>

  <artifactId>jacoco-maven-plugin</artifactId>
  <packaging>maven-plugin</packaging>

  <name>JaCoCo :: Maven Plugin</name>
  <description>The JaCoCo Maven Plugin provides the JaCoCo runtime agent to your tests and allows basic report creation.</description>

  <prerequisites>
    <maven>2.2.1</maven>
  </prerequisites>

  <dependencies>
    <dependency>
      <groupId>org.apache.maven</groupId>
      <artifactId>maven-plugin-api</artifactId>
      <version>${project.prerequisites.maven}</version>
    </dependency>
    <dependency>
      <groupId>org.apache.maven</groupId>
      <artifactId>maven-project</artifactId>
      <version>${project.prerequisites.maven}</version>
    </dependency>
    <dependency>
      <groupId>org.codehaus.plexus</groupId>
      <artifactId>plexus-utils</artifactId>
      <version>1.5.6</version>
    </dependency>
    <dependency>
      <groupId>org.apache.maven.shared</groupId>
      <artifactId>file-management</artifactId>
      <version>1.2.1</version>
    </dependency>

    <dependency>
      <groupId>org.apache.maven.reporting</groupId>
      <artifactId>maven-reporting-api</artifactId>
      <version>${project.prerequisites.maven}</version>
    </dependency>
    <dependency>
      <groupId>org.apache.maven.reporting</groupId>
      <artifactId>maven-reporting-impl</artifactId>
      <version>2.1</version>
    </dependency>

    <dependency>
      <groupId>${project.groupId}</groupId>
      <artifactId>org.jacoco.agent</artifactId>
      <classifier>runtime</classifier>
    </dependency>
    <dependency>
      <groupId>${project.groupId}</groupId>
      <artifactId>org.jacoco.core</artifactId>
    </dependency>
    <dependency>
      <groupId>${project.groupId}</groupId>
      <artifactId>org.jacoco.report</artifactId>
    </dependency>
  </dependencies>

  <build>
    <sourceDirectory>src</sourceDirectory>
    <resources>
      <resource>
        <directory>META-INF</directory>
        <targetPath>META-INF</targetPath>
      </resource>
    </resources>

    <plugins>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-javadoc-plugin</artifactId>
        <configuration>
          <tagletArtifacts>
            <tagletArtifact>
              <groupId>org.apache.maven.plugin-tools</groupId>
              <artifactId>maven-plugin-tools-javadoc</artifactId>
              <version>2.8</version>
            </tagletArtifact>
          </tagletArtifacts>
        </configuration>
      </plugin>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-plugin-plugin</artifactId>
        <executions>
          <execution>
            <id>help-mojo</id>
            <goals>
              <goal>helpmojo</goal>
            </goals>
          </execution>
          <execution>
            <id>report</id>
            <phase>package</phase>
            <goals>
              <goal>report</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
    </plugins>
  </build>
</project>
```

Any thoughts?

**Comment by pdurbin** [comment]
@bricas thanks for the heads up about this. I see a similar [elsewhere](http://webcache.googleusercontent.com/search?q=cache:Esqa00JVbxgJ:stackoverflow.com/questions/33308435/orientdb-build-fails-with-weird-jacoco-version+&cd=1&hl=en&ct=clnk&gl=us).

I just tested the 2731-techupdate branch yesterday as part of #2151 and artifacts should be downloaded fresh in an ~/.m2 directory on the VM.

Yesterday a collaborator at @sbgrid mentioned Vagrant couldn't build war from 36128595c077d8b46b5d890d9e980209cba264c1 which is what we tagged as 4.2.1: https://github.com/IQSS/dataverse/tree/v4.2.1

Where there's smoke there's fire...

As a workaround you can grab the war file from https://github.com/IQSS/dataverse/releases/download/v4.2.1/dataverse-4.2.1.war so that Vagrant doesn't have to built it but let's leave this issue open until we've confirmed that Vagrant is still able to build a war file.

I would start with `vagrant destroy` followed by `vagrant up` in case this was a transient problem.

**Comment by pdurbin** [comment]
I switched to the v4.2.1 tag and I'm having trouble with `vagrant up` too. It seems to be a different problem though. I can't install anything via yum, such as postgres:

```
==> standalone: Adding Shibboleth yum repo
==> standalone: Loaded plugins: fastestmirror, security
==> standalone: Determining fastest mirrors
==> standalone:  * base: mirrors.syringanetworks.net
==> standalone:  * extras: mirror.cogentco.com
==> standalone:  * updates: mirrors.syringanetworks.net
==> standalone: http://yum.puppetlabs.com/el/6/dependencies/x86_64/repodata/587eb46d40fc0ffaa936fa1154fedcc31c65b5b2-primary.sqlite.bz2: [Errno 14] PYCURL ERROR 22 - "The requested URL returned error: 404 Not Found"
==> standalone: Trying other mirror.
==> standalone: http://yum.puppetlabs.com/el/6/dependencies/x86_64/repodata/587eb46d40fc0ffaa936fa1154fedcc31c65b5b2-primary.sqlite.bz2: [Errno 14] PYCURL ERROR 22 - "The requested URL returned error: 404 Not Found"
==> standalone: Trying other mirror.
==> standalone: Error: failure: repodata/587eb46d40fc0ffaa936fa1154fedcc31c65b5b2-primary.sqlite.bz2 from puppetlabs-deps: [Errno 256] No more mirrors to try.
==> standalone: postgresql: unrecognized service
```

Hopefully these are transient problems. I just added to my growing list of automated testing ideas at #2746 the idea that we should test `vagrant up` on a server (not sure how to test that it worked or not, though).

I'm going to try again in a new branch in a bit related to #2628 and I'll report back.

**Comment by bricas** [comment]
I was getting that same error just now too -- but it seems to have been resolved.

FYI, I tried to destroy and re-up a number of times yesterday (into this morning) and had the exact same issue i initially reported.

I'm currently attempting to ```vagrant up``` with the pre-baked war file. Will report back when it finishes.

**Comment by bricas** [comment]
Success!

**Comment by pdurbin** [comment]
@bricas great. It worked just now on the 2731-techupdate branch. Dunno. Lots of moving parts behind that single `vagrant up` command, of course. I'm going to leave this open for a bit until I hear back from @sbgrid.

**Comment by pdurbin** [comment]
Ok, I messed around with Vagrant a bit more in a new 2628-payara branch (forked from 2731-techupdate branch) and the Vagrant part seems to work fine: https://github.com/IQSS/dataverse/commit/1c0c7be6ebc332a1fd62c7d09cb09badb18d0c41

Since @bricas says Vagrant is now working on the v4.2.1 tag I'm going to close this issue. (This was blessed by @kcondon .) Maybe it was a transient problem, my favorite. @bricas if you have ideas for making this the Vagrant setup more robust, please let me know. People like @posixeleni sometimes use it such as when she is editing tsv files for metadata blocks.

I have an email thread going with @sbgrid so I'll ask them to reopen this issue or create a new one if the problem they're seeing is different.

**Comment by bricas** [comment]
My vagrant only worked with the pre-built .war file Without it, the build step failed every time in the exact same way as originally reported.

**Comment by pdurbin** [comment]
@bricas hmm. Let me re-open this issue then, and think about it some more.

**Comment by pdurbin** [comment]
I just tried `vagrant up` in the 4.2.2 branch (commit 4a3e7b2) and I'm getting the same failure that @bricas originally reported:

`BUILD FAILURE`

...

` [ERROR] Plugin org.jacoco:jacoco-maven-plugin:0.7.5.201505241946 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.jacoco:jacoco-maven-plugin:jar:0.7.5.201505241946: Could not transfer artifact org.jacoco:jacoco-maven-plugin:pom:0.7.5.201505241946 from/to central (https://repo.maven.apache.org/maven2): java.security.ProviderException: java.security.KeyException -> [Help 1]`

Here's the output: [vagrant-up-fail-4a3e7b2.txt](https://github.com/IQSS/dataverse/files/47429/vagrant-up-fail-4a3e7b2.txt)

Vagrant is really only used for demonstration and testing purposes but I'm going to go ahead and put this in the 4.2.2 milestone since I'm having trouble in that branch. Heads up to @scolapasta 

As I said earlier at https://github.com/IQSS/dataverse/issues/2744#issuecomment-157785373 I believe `vagrant up` is working in the 2731-techupdate branch. Very strange.

**Comment by pdurbin** [comment]
Interesting... `yum install nss` seems to be helping. I found this tip here: MESOS-3752 CentOS 6 dependency install fails at Maven - ASF JIRA - https://issues.apache.org/jira/browse/MESOS-3752

**Comment by pdurbin** [comment]
In ef1720b I added `yum install nss` and now Vagrant is able to build the war file again. I don't really understand why nss is suddenly necessary but looking at https://issues.apache.org/jira/browse/MESOS-3752 might provide a clue.

@bricas thanks for pointing out that Vagrant was failing to build the war file (I updated the subject of this issue). I made my fix in the 4.2.2 branch so when we tag it, people should be able to get `vagrant up` working from the latest tag. I'm going to go ahead and close this issue.
